core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/predicate/UnaryPredicateOperator.java

```diff
@@ -82,4 +106,62 @@ private static FunctionResolver isNotNull() {
             .collect(
                 Collectors.toList()));
   }
+
+  private static FunctionResolver ifNull() {
+    FunctionName functionName = BuiltinFunctionName.IFNULL.getName();
+    List<ExprType> typeList = ExprCoreType.coreTypes();
+    typeList.add(UNKNOWN);
+
+    List<SerializableFunction<FunctionName, org.apache.commons.lang3.tuple.Pair<FunctionSignature,
+            FunctionBuilder>>> functionsOne = typeList.stream().map(v ->
+            impl((UnaryPredicateOperator::exprIfNull), v, v, v))
+            .collect(Collectors.toList());
+
+    List<SerializableFunction<FunctionName, org.apache.commons.lang3.tuple.Pair<FunctionSignature,
+            FunctionBuilder>>> functionsTwo = typeList.stream().map(v ->
+            impl((UnaryPredicateOperator::exprIfNull), v, UNKNOWN, v))
+            .collect(Collectors.toList());
+
+    functionsOne.addAll(functionsTwo);
+    FunctionResolver functionResolver = FunctionDSL.define(functionName, functionsOne);
+    return functionResolver;
+  }
+
+  private static FunctionResolver nullIf() {
+    FunctionName functionName = BuiltinFunctionName.NULLIF.getName();
+    List<ExprType> typeList = ExprCoreType.coreTypes();
+    typeList.add(UNKNOWN);
+
+    FunctionResolver functionResolver =
+        FunctionDSL.define(functionName,
+            typeList.stream().map(v ->
+              impl((UnaryPredicateOperator::exprNullIf), v, v, v))
+              .collect(Collectors.toList()));
+    return functionResolver;
+  }
+
+  /** v2 if v1 is null.
+   * @param v1 varable 1
+   * @param v2 varable 2
+   * @return v2 if v1 is null
+   */
+  public static ExprValue exprIfNull(ExprValue v1, ExprValue v2) {
+    return (v1.isNull() || v1.isMissing()) ? v2 : v1;
+  }
+
+  /** null if v1 equls to v2.
+   * @param v1 varable 1
+   * @param v2 varable 2
+   * @return null if v1 equls to v2
+   */
+  public static ExprValue exprNullIf(ExprValue v1, ExprValue v2) {
+    if (v1.isNull() || v1.isMissing() || v2.isNull() || v2.isMissing()) {
+      return v1;
+    } else if (v1.value().equals(v2.value())) {
```

Comment:
could we use the v1.equals(v2) directly?

Reply:
yes, thx

Reply:
public class Main {
  public static void main(String[] args) {
    String myStr1 = null;
    System.out.println("debug : " + myStr1.equals("test"));
  }
}
Exception in thread "main" java.lang.NullPointerException	at Main.main(Main.java:4)
if v1 is null, using the v1.equals(v2) directly will cause exception.